test/source/tests/settings.ts

```diff
@@ -706,6 +706,45 @@ export const defineSettingsTests = (testVariant: TestVariant, testWithBrowser: T
         internalTestState.expectIntentionalErrReport = true;
       })
     );
+    test(
+      'settings - Errors encountered during the addition/update of a private key should be handled properly',
+      testWithBrowser(async (t, browser) => {
+        t.context.mockApi!.configProvider = new ConfigurationProvider({
+          attester: {
+            pubkeyLookup: {},
+          },
+        });
+        const acct = 'flowcrypt.compatibility@gmail.com';
+        const settingsPage = await BrowserRecipe.openSettingsLoginApprove(t, browser, acct);
+        const keyCanBeFixedTestKey = testConstants.keyCanBeFixedTestKey;
+        await SetupPageRecipe.manualEnter(settingsPage, '', {
+          fixKey: true,
+          key: {
+            title: '',
+            armored: keyCanBeFixedTestKey,
+            passphrase: 'hard to guess passphrase',
+            longid: '9866DB9063926D73',
+          },
+        });
+        await SettingsPageRecipe.toggleScreen(settingsPage, 'additional');
+        await settingsPage.waitAndClick('@action-open-pubkey-page');
+        const myKeyPage = await settingsPage.getFrame(['my_key.htm']);
+        await myKeyPage.waitAndClick('@action-update-prv');
+        await myKeyPage.waitAndClick('@source-paste');
+        const invalidKeyFormat = 'non-valid-private-key-format';
+        const invalidPrivateKeyBlock = '-----BEGIN PGP PRIVATE KEY BLOCK-----\r\n\r\ninvalid-key\r\n-----END PGP PRIVATE KEY BLOCK-----';
+        await myKeyPage.waitAndType('@input-prv-key', invalidKeyFormat);
+        await myKeyPage.waitAndClick('@action-update-key');
+        await myKeyPage.waitAndRespondToModal('warning', 'confirm', 'This does not appear to be a validly formatted key.');
+        await myKeyPage.waitAndType('@input-prv-key', invalidPrivateKeyBlock);
+        await myKeyPage.waitAndClick('@action-update-key');
+        await myKeyPage.waitAndRespondToModal('error', 'confirm', 'An error happened when processing the key');
+        await myKeyPage.waitAndType('@input-prv-key', keyCanBeFixedTestKey);
+        await myKeyPage.waitAndClick('@action-update-key');
```

Comment:
on this step we get error, but don't check it:

<img width="683" alt="Screenshot 2023-09-19 at 14 34 13" src="https://github.com/FlowCrypt/flowcrypt-browser/assets/1062093/adbd2ea7-fd89-40bd-91e6-fb0271aa37e7">

Reply:
Ah, yes. This particular error is handle differently in the `my_key_update.htm`. For instance, during setup, the FlowCrypt browser extension offer the user to correct their key by updating the user id wherein during update it will throw this error. 

My opinion here is 50/50 -- the key can be updated here, though if they use a "fixable" key during setup, then it will be fixed in that point where they have to update the key because the updated key during setup can be downloadable.

Reply:
I think we can leave current implementation for now, as this PR is aimed at adding tests of errors handling, so here we should just check correct error message.

Reply:
Okay - thanks, I have also checked for the correct error message.

Reply:
Yeah, looks good now 👍